tests/conftest.py

```diff
@@ -446,14 +446,22 @@ def reset_all_objects():
     unknown_parameters.clear()
 
 
+@pytest.fixture(scope="session", autouse=True)
+def init_cwd():
+    """
+    Fixture returning the cwd at the beginning of the session.
+    """
+    yield os.getcwd()
+
+
 @pytest.fixture(scope="function", autouse=True)
-def restore_cwd():
+def restore_cwd(init_cwd: str):
     """
-    Restore the current working directory after search test.
+    Restore the current working directory after search test. The cwd should be restore using the init_cwd fixture
+    to prevent that another function scoped fixture overrides the cwd before the initial cwd has been saved.
```

Comment:
Which fixture caused the issue here? As far as I know autouse fixtures take precedence so it'd have to be another autouse fixture.

Reply:
Autouse fixtures only takes precedence within their scope. Since `restore_cwd` is a function scoped fixure, all session scoped fixtures take precedence as well. The problem was caused by the `project` fixture of pytest-inmanta. This fixture relies on the `project_shared` fixture, which is a session scoped fixture. The latter fixture calls `module.Project.set(test_project)` (at some time). This method executes `os.chdir(project._path)`.